File: src/main/java/com/uitnetwork/alexa/service/SpeechletResponseService.java
package com.uitnetwork.alexa.service;

import org.springframework.stereotype.Service;

import com.amazon.speech.speechlet.SpeechletResponse;
import com.amazon.speech.ui.PlainTextOutputSpeech;
import com.amazon.speech.ui.Reprompt;
import com.amazon.speech.ui.SsmlOutputSpeech;

@Service
public class SpeechletResponseService {

    public SpeechletResponse newAskResponse(String message, String repromptMessage) {
        PlainTextOutputSpeech plainTextOutputSpeech = new PlainTextOutputSpeech();
        plainTextOutputSpeech.setText(message);

        PlainTextOutputSpeech repromptOutputSpeech = new PlainTextOutputSpeech();
        repromptOutputSpeech.setText(repromptMessage);
        Reprompt reprompt = new Reprompt();
        reprompt.setOutputSpeech(repromptOutputSpeech);

        return SpeechletResponse.newAskResponse(plainTextOutputSpeech, reprompt);
    }

    public SpeechletResponse newTellResponse(String ssmlMessage) {
        SsmlOutputSpeech ssmlOutputSpeech = new SsmlOutputSpeech();
        ssmlOutputSpeech.setSsml("<speak>" + ssmlMessage + "</speak>");

        return SpeechletResponse.newTellResponse(ssmlOutputSpeech);
    }
}

File: src/main/kotlin/com/uitnetwork/alexa/service/SpeechletResponseService.kt
package com.uitnetwork.alexa.service

import com.amazon.speech.speechlet.SpeechletResponse
import com.amazon.speech.ui.PlainTextOutputSpeech
import com.amazon.speech.ui.Reprompt
import com.amazon.speech.ui.SsmlOutputSpeech
import org.springframework.stereotype.Service

@Service
class SpeechletResponseService {

    fun newAskResponse(message: String, repromptMessage: String): SpeechletResponse {
        val plainTextOutputSpeech = PlainTextOutputSpeech()
        plainTextOutputSpeech.text = message

        val repromptOutputSpeech = PlainTextOutputSpeech()
        repromptOutputSpeech.text = repromptMessage
        val reprompt = Reprompt()
        reprompt.outputSpeech = repromptOutputSpeech

        return SpeechletResponse.newAskResponse(plainTextOutputSpeech, reprompt)
    }

    fun newTellResponse(ssmlMessage: String): SpeechletResponse {
        val ssmlOutputSpeech = SsmlOutputSpeech()
        ssmlOutputSpeech.ssml = "<speak>$ssmlMessage</speak>"

        return SpeechletResponse.newTellResponse(ssmlOutputSpeech)
    }
}
